src/data/debts.tsx

```diff
@@ -0,0 +1,178 @@
+  const [user] = useRecoilState(userState);
+  const [cookies] = useCookies(["jwt"]);
+
+  const getDebts = async () => {
+    try {
+      if (user) {
+        const response = await axios.get(`${port}/debts/${user?._id}`, {
+          headers: {
+            Authorization: "Bearer " + cookies.jwt,
+          },
+        });
+        if (response.status === 200) {
+          setDebtsList(response.data);
+        } else {
+          createToastify({ text: "Debt or Loan not found", type: "error" });
+        }
+      }
+    } catch (err: unknown) {
+      if (err instanceof AxiosError) {
+        createToastify({
+          text: err.response?.data.message || err.message,
+          type: "error",
+        });
+        throw new Error(err.message);
+      }
+    }
+  };
+
+  const addDebt = async (newItem: DebtFormI) => {
+    try {
+      if (user) {
+        const parsedItem = {
+          ...newItem,
+          amount: parseInt(newItem.amount),
+          userId: user._id,
+        };
+        const response = await axios.post(`${port}/debts/`, parsedItem, {
+          headers: {
+            Authorization: "Bearer " + cookies.jwt,
+          },
+        });
+        if (response.status === 200) {
+          setDebtsList([...debtsList, response.data]);
+        } else {
+          createToastify({
+            text: "Add Debt not successful",
+            type: "error",
+          });
+        }
+      }
+    } catch (err: unknown) {
+      if (err instanceof AxiosError) {
+        createToastify({
+          text: err.response?.data.message || err.message,
+          type: "error",
+        });
+        throw new Error(err.message);
+      }
+    }
+  };
+
+  const editDebt = async (id: string, updatedItem: DebtFormI) => {
+    try {
+      if (user) {
+        const parsedItem = {
+          ...updatedItem,
+          amount: parseInt(updatedItem.amount),
+          userId: user._id,
+        };
+        const response = await axios.put(`${port}/debts/${id}`, parsedItem, {
+          headers: {
+            Authorization: "Bearer " + cookies.jwt,
+          },
+        });
+        if (response.status === 200) {
+          const i = debtsList.findIndex((elem) => elem._id === id);
+          if (i !== -1) {
+            setDebtsList([
+              ...debtsList.slice(0, i),
+              response.data,
+              ...debtsList.slice(i + 1),
+            ]);
+          } else {
+            throw new Error("ID not found updating debts list");
+          }
+        } else {
+          createToastify({
+            text: "Edit Debt not successful",
+            type: "error",
+          });
+        }
+      }
+    } catch (err: unknown) {
+      if (err instanceof AxiosError) {
+        createToastify({
+          text: err.response?.data.message || err.message,
+          type: "error",
+        });
+        throw new Error(err.message);
+      }
+    }
+  };
+
+  const deleteDebt = async (id: string) => {
+    try {
+      if (user) {
```

Comment:
If there is not user. How are we handling that? Should be good to handle that error, in the case somethign strange happen with that